### PR TITLE
Bug fix for reverse sequence cursor. Not sufficient to just mark a cu…

### PR DESCRIPTION
**Pull request by zachmu**
…rsor reverse after the fact, need to initialize cursors at all levels of the tree as reverse as well. Still needs tests, but fixes panics in sqllogictest.

Signed-off-by: Zach Musgrave <zach@liquidata.co>

### CODE REVIEW

**Comment by zachmu** [comment]
Still needs unit tests, but does this seem like a reasonable solution? Fixes panics in sqllogictest.

**Comment by zachmu** [comment]
Aaron, PTAL

File: go/store/types/ordered_sequences.go [hunk 1]
```diff
@@ -89,6 +89,72 @@ func newCursorAt(ctx context.Context, seq orderedSequence, key orderedKey, forIn
...
+		if cs == nil {
+			break
+		}
+		seq = cs.(orderedSequence)
+	}
+
+	// If we overshot the key, back off by one. We want the greatest element <= key
+	if cur.valid() {
+		currKey, err := cur.seq.(orderedSequence).getKey(cur.idx)
+		d.PanicIfError(err)
```
Comment (reltuk) [inline go/store/types/ordered_sequences.go:139]: Is this an appropriate panic, or should be return err?

Comment (zachmu) [reply go/store/types/ordered_sequences.go]: Good call

**Pull request merged**